### PR TITLE
Configurable riak buckets

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR allows to configure bucket names.

Proposed changes include:
* configurations options
* docs

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7308.1](https://travis-ci.org/esl/MongooseIM/jobs/597627718) / Erlang 22.0 / small_tests / 6ccce67fc7ec765f023ed6889a736ea3c42f01ea
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2490/7308/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2490/7308/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-14_13.26.38/lib.mongooseim.logs/run.2019-10-14_13.26.39/suite.log.html)

File: src/mod_roster_riak.erl [hunk 1]
```diff
@@ -32,8 +32,8 @@
          del_roster_t/3,
          raw_to_record/2]).
 
--define(ROSTER_BUCKET(LServer), {<<"rosters">>, LServer}).
--define(VER_BUCKET(LServer), {<<"roster_versions">>, LServer}).
+-define(ROSTER_BUCKET(LServer), {gen_mod:get_module_opt(LServer, mod_roster, bucket_type, <<"rosters">>), LServer}).
+-define(VER_BUCKET(LServer), {gen_mod:get_module_opt(LServer, mod_roster, version_bucket_type, <<"roster_versions">>), LServer}).
```
Comment (NelsonVides) [inline src/mod_roster_riak.erl]: These are some seriously long line,s keep it below a 100 characters please xD

**Comment by mongoose-im** [comment]
[7346.1](https://travis-ci.org/esl/MongooseIM/jobs/599228300) / Erlang 22.0 / small_tests / 878ffa1fd77a0b301fe345e1564c7e665a60604f
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2490/7346/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2490/7346/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-10-17_16.46.22/lib.mongooseim.logs/run.2019-10-17_16.46.22/suite.log.html)

File: doc/authentication-backends/Riak-authentication-module.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+* **bucket_type:**
+    * **Description:** Riak bucket type.
+    * **Value:** Binary
+    * **Default:** `<<"users">>`
+
+Example:
+
+```erlang
+{auth_opts, [
+ {bucket_type, <<"users">>}
```
Comment (NelsonVides) [inline doc/authentication-backends/Riak-authentication-module.md]: Great! :smile:

**Comment by michalwski** [comment]
> Wait, if we're adding an auth backend file, shouldn't that also be added here:
> 
> https://github.com/esl/MongooseIM/blob/ac08e4f37fdd1c4bcb22b5e7d27d459fd4ec74c0/mkdocs.yml#L49
> 
> 🤔

You are right, good catch @NelsonVides

**Comment by michalwski** [comment]
@NelsonVides @arcusfelis actually it's not required to add it to the `mkdocs` unless we want this page in the nav bar. This page is accessible form the `Advanced configuration`, there is a link to `riak` auth backend. What do you think?

**Comment by NelsonVides** [comment]
> actually it's not required to add it to the `mkdocs` unless we want this page in the nav bar. This page is accessible form the `Advanced configuration`, there is a link to `riak` auth backend. What do you think?

Yeah, this is kinda subjective, I don't have like a very strong opinion. But I kinda like having a comprehensive link index (nav bar).

**Pull request merged**